php/commands/src/CLI_Alias_Command.php

```diff
@@ -0,0 +1,335 @@
+ *
+ *     # Update alias.
+ *     $ wp cli alias update @prod newlogin@host /updated/wordpress/path/
+ *     Success: Updated 'prod' alias.
+ *
+ *     # Delete alias.
+ *     $ wp cli alias delete @prod
+ *     Success: Deleted '@prod' alias.
+ *
+ * @package wp-cli
+ * @when before_wp_load
+ */
+class CLI_Alias_Command extends WP_CLI_Command {
+
+	/**
+	 * List available WP-CLI aliases.
+	 *
+	 * Aliases are shorthand references to WordPress installs. For instance,
+	 * `@dev` could refer to a development install and `@prod` could refer to
+	 * a production install. This command gives you visibility in what
+	 * registered aliases you have available.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: yaml
+	 * options:
+	 *   - yaml
+	 *   - json
+	 *   - var_export
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # List all available aliases.
+	 *     $ wp cli alias list
+	 *     ---
+	 *     @all: Run command against every registered alias.
+	 *     @prod:
+	 *       ssh: runcommand@runcommand.io~/webapps/production
+	 *     @dev:
+	 *       ssh: vagrant@192.168.50.10/srv/www/runcommand.dev
+	 *     @both:
+	 *       - @prod
+	 *       - @dev
+	 *
+	 * @subcommand list
+	 */
+	public function list_( $args, $assoc_args ) {
+		WP_CLI::print_value( WP_CLI::get_runner()->aliases, $assoc_args );
+	}
+
+	/**
+	 * Gets the value for an alias.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <key>
+	 * : Key for the alias.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get alias.
+	 *     $ wp cli alias get @prod
+	 *     ssh: dev@somedeve.env:12345/home/dev/
+	 *
+	 * @subcommand get
+	 */
+	public function get( $args, $assoc_args ) {
+		list( $alias ) = $args;
+
+		$aliases = WP_CLI::get_runner()->aliases;
+
+		if ( ! empty( $aliases[ $alias ] ) ) {
+			foreach ( $aliases[ $alias ] as $key => $value ) {
+				WP_CLI::log( sprintf( ' %1$s: %2$s', $key, $value ) );
+			}
+		} else {
+			WP_CLI::error( sprintf( 'No alias found with key \'%s\'.', $alias ) );
+		}
+	}
+
+	/**
+	 * Creates an alias.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <key>
+	 * : Key for the alias.
+	 *
+	 * <login>
+	 * : SSH Login.
+	 *
+	 * <wp_path>
+	 * : WordPress install path.
+	 *
+	 * [--config=<config>]
+	 * : Config file to be considered for operations.
+	 * ---
+	 * default: global
+	 * options:
+	 *   - global
+	 *   - project
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Add alias to global config.
+	 *     $ wp cli alias add @prod login@host /path/to/wordpress/install/
+	 *     Success: Added '@prod' alias.
+	 *
+	 *     # Add alias to project config.
+	 *     $ wp cli alias add @prod login@host /path/to/wordpress/install/ --config=project
+	 *     Success: Added '@prod' alias.
+	 *
+	 * @subcommand add
+	 */
+	public function add( $args, $assoc_args ) {
+
+		$config = ( ! empty( $assoc_args['config'] ) ? $assoc_args['config'] : 'global' );
+
+		list( $config_path, $aliases ) = $this->_get_aliases_data( $config, '' );
```

Comment:
No need for private methods to be prefixed with an underscore, that is PHP 4 code style.
```suggestion
		list( $config_path, $aliases ) = $this->get_aliases_data( $config, '' );
```